sqlglot/dialects/bigquery.py

```diff
@@ -229,6 +229,44 @@ def _build_time(args: t.List) -> exp.Func:
     return exp.Anonymous(this="TIME", expressions=args)
 
 
+# Ref(values): https://cloud.google.com/bigquery/docs/reference/standard-sql/date_functions#extract
+# Ref (keys): https://docs.snowflake.com/en/sql-reference/functions-date-time#supported-date-and-time-parts
+# Ref (keys): https://docs.databricks.com/en/sql/language-manual/functions/extract.html#returns
+EXTRACT_PART_MAPPING = {
```

Comment:
i'd like this to be done in a bit more generic way. right now, only bigquery has the ability to transpile snowflake/spark to itself. instead, for every dialect, we need to convert the part into a sqlglot variant, so for example

dd -> day, days -> day.

this may be good enough, then every dialect will be normalized.

if we want to go further we can define gen overrides, so for example sqlglot day -> DAYS if we wanted to, but that may not be necessary.

so to reiterate, add some base parser / dialect mapping all variants to a sqlglot normalized variant. if necessary (maybe it's not because there's no need), allow dialects to define a mapping from sqlglot variant to it's own variant.

Reply:
Makes sense, that's definitely a better solution. It'll take me a bit longer to do this. Is it ok that this means that some transpilations won't be an identity? (e.g., `transpile("EXTRACT(dd FROM x)", read="snowflake", write="snowflake")` -> `EXTRACT(day FROM x)`

Reply:
correct, it's never been a contract to produce identities. for exmaple

select x::int -> select (cast x as int)

Reply:
Hey @maureen-daum 👋

Just wanted to double-check: is this a WIP on your end?

Reply:
Oh yup it is! I'll mark as a draft (or I can close and just open a new pr once it's ready)

Reply:
@maureen-daum I'll go ahead and close this PR for now, let's iterate on a new one when it's ready.